app/serializers/admin_user_action_serializer.rb

```diff
@@ -80,4 +80,15 @@ def action_type
       .select { |ua| [UserAction::REPLY, UserAction::RESPONSE].include? ua.action_type }
       .first.try(:action_type)
   end
+
+  # we need this to handle deleted topics which aren't loaded via
+  # Topic.unscoped do
+  #   Post.includes(:topic)
+  # end
+  # because Rails 4 "unscoped" support is still bugged (cf. https://github.com/rails/rails/issues/13775)
+  def topic
+    return @topic if @topic
+    @topic = object.topic || Topic.with_deleted.find(object.topic_id)
```

Comment:
I am not a fan of doing SQL queries because it can lead to N+1, but I believe in this case it is fine.

Reply:
Ackchyually... there may be a better way if we modify https://github.com/discourse/discourse/blob/6f4cf814013c193dd83dd0f44c53a3147f382e8a/app/controllers/posts_controller.rb#L712-L734 altogether. I think I can spend more time on that method since it's only used in `deleted_posts` and `flagged_posts`.

Reply:
Since it is an admin only serializer, I do not think it will have a huge impact. I believe we are good to go now and we can refine at a later stage if it causes any issues.